File: Baekjoon/Class4/10.1629 곱셈.py
import sys
input = sys.stdin.readline

'''
a^n
= a if n==1,
= (a^(n/2))^2 if n%2==0,
= a*(a^((n-1)/2))^2 if n%2==1
'''

def power_mod(base, exp, div):
    if (exp == 1): return base % div #1일 때
    half = power_mod(base, exp // 2, div)
    whole = (half * half) % div
    if (exp % 2 == 0): return whole; #짝수일 때
    return ((base % div) * whole) % div #홀수일 때

a, b, c = map(int, input().split())
print(power_mod(a, b, c))

File: Baekjoon/Class4/10.1629 곱셈.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

/*
a^n
= a if n==1,
= (a^(n/2))^2 if n%2==0,
= a*(a^((n-1)/2))^2 if n%2==1
*/

const powerMod = (base, exp, div) => {
    if(exp === 1n) return base % div; // 1일 때
    const half = powerMod(base, exp / 2n, div);
    const whole = (half * half) % div;
    if(exp % 2n === 0n) return whole; // 짝수일 때
    return ((base % div) * whole) % div; // 홀수일 때
};

const [a, b, c] = input.split(" ").map(BigInt);
console.log(Number(powerMod(a, b, c)));